_sources/Iteration/Exercises.rst

```diff
@@ -14,31 +14,139 @@
+
+
+           .. actex:: ex_3_3
+
+              Write a program that uses a for loop to print
+                 |  ``One of the months of the year is January``
+                 |  ``One of the months of the year is February``
+                 |  ``One of the months of the year is March``
+                 |  etc ...
+              ~~~~
+
+        .. tab:: Answer
+
+            .. activecode:: q3_answer
+
+
+                for amonth in ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'November', 'December']:
+                    print("One of the months of the year is", amonth)
+
+        .. tab:: Discussion
+
+            .. disqus::
+                :shortname: interactivepython
+                :identifier: b271442ee0864973a023c19f27aeb401
+
+
+.. question:: iter_ex_4
+
+
+   .. actex:: ex_3_4
+
+      Assume you have a list of numbers ``12, 10, 32, 3, 66, 17, 42, 99, 20``
+
+      a. Write a loop that prints each of the numbers on a new line.
+      b. Write a loop that prints each number and its square on a new line.
+      ~~~~
+
+.. question:: iter_ex_6
+
+   .. actex:: ex_3_6
+      :nocodelens:
+
+      Write a program that asks the user for the number of sides, the length of the side, the color, and the fill color of a
+      regular polygon.  The program should draw the polygon and then fill it in.
+      ~~~~
+
+
+.. question:: iter_ex_7
+
+   .. tabbed:: q7
+
+       .. tab:: Question
+
+            .. actex:: ex_3_7
+               :nocodelens:
+
+               A drunk pirate makes a random turn and then takes 100 steps forward, makes another random turn, takes another 100 steps, turns another random amount, etc.  A social science student records the angle of each turn before the next 100 steps are taken.  Her experimental data is ``160, -43, 270, -97, -43, 200, -940, 17, -86``. (Positive angles are counter-clockwise.)  Use a turtle to draw the path taken by our drunk friend.  After the pirate is done walking, print the current heading.
+               ~~~~
+
+       .. tab:: Answer
+
+           .. activecode:: iter_q7_answer
+               :nocodelens:
+
+               import turtle
+
+               wn = turtle.Screen()
+               lovelace = turtle.Turtle()
+
+               # move the turtle forward a little so that the whole path fits on the screen
+               lovelace.penup()
+               lovelace.forward(60)
+
+               # now draw the drunk pirate's path
+               lovelace.pendown()
+               for angle in [160, -43, 270, -97, -43, 200, -940, 17, -86]:
+
+                   # we use .left() so that positive angles are counter-clockwise
+                   # and negative angles are clockwise
+                   lovelace.left(angle)
+                   lovelace.forward(100)
+
+               # the .heading() method gives us the turtle's current heading in degrees
+               print("The pirate's final heading was", lovelace.heading())
```

Comment:
Oh, I didn't realize there was a heading() method. I'm not sure we ever told them about that. How about adding a hint that the initial heading is <whatever it is> and then have them accumulate all the angle changes?

Reply:
Sure having them accumulate the heading changes would be good.

Reply:
As a bonus they will need to use the modulo operator to have the angle between 0-360